Sources/NIOCore/Docs.docc/loops-futures-concurrency.md

```diff
@@ -0,0 +1,176 @@
+actor.
+
+## Event Loops
+
+NIO's core execution primitive is the ``EventLoop``. An ``EventLoop`` is fundamentally nothing more than
+a Swift Concurrency Serial Executor that can also perform I/O operations directly. Indeed, NIO's
+``EventLoop``s can be exposed as serial executors, using ``EventLoop/executor``. This provides a mechanism
+to protect actor-isolated state using a NIO event-loop. With [the introduction of task executors](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0417-task-executor-preference.md),
+future versions of SwiftNIO will also be able to offer their event loops for individual `Task`s to execute
+on as well.
+
+In a Swift 6 world, it is possible that these would be the API that NIO offered to execute tasks on the
+loop. However, as NIO predates Swift 6, it also offers its own set of APIs to enqueue work. This includes
+(but is not limited to):
+
+- ``EventLoop/execute(_:)``
+- ``EventLoop/submit(_:)``
+- ``EventLoop/scheduleTask(in:_:)``
+- ``EventLoop/scheduleRepeatedTask(initialDelay:delay:notifying:_:)``
+- ``EventLoop/scheduleCallback(at:handler:)-2xm6l``
+
+The existence of these APIs requires us to also ask the question of where the submitted code executes. The
+answer is that the submitted code executes on the event loop (or, in Swift Concurrency terms, on the
+executor provided by the event loop).
+
+As the event loop only ever executes a single item of work (either an `async` function or one of the
+closures above) at a time, it is a _serial_ executor. It also provides an _isolation domain_: code
+submitted to a given `EventLoop` never runs in parallel with other code submitted to the same loop.
+
+The result here is that a all closures passed into the event loop to do work must be transferred
+in: they may not be kept hold of outside of the event loop. That means they must be sent using
+the `sending` keyword.
+
+> Note: As of the current 2.75.0 release, NIO enforces the stricter requirement that these closures
+    are `@Sendable`. This is not a long-term position, but reflects the need to continue
+    to support Swift 5 code which requires this stricter standard. In a future release of
+    SwiftNIO we expect to relax this constraint: if you need this relaxed constraint
+    then please file an issue.
+
+## Event loop futures
+
+In Swift NIO the most common mechanism to arrange a series of asynchronous work items is
+_not_ to queue up a series of ``EventLoop/execute(_:)`` calls. Instead, users typically
+use ``EventLoopFuture``.
+
+``EventLoopFuture`` has some extensive semantics documented in its API documentation. The
+most important principal for this discussion is that all callbacks added to an
+``EventLoopFuture`` will execute on the ``EventLoop`` to which that ``EventLoopFuture`` is
+bound. By extension, then, all callbacks added to an ``EventLoopFuture`` execute on the same
+executor (the ``EventLoop``) in the same isolation domain.
```

Comment:
I would love if we could highlight that part more. I am always thinking of our ELs as both an executor but also as an isolation domain.

Reply:
Line 54 has this earlier in the doc, is there a step you think I should use to punch it up?

Reply:
Oh damn I somehow read right over the second part of that sentence. Ignore me.